File: CI Codes/MapReduce/Without Hadoop/word_count.py
import sys
import re
from collections import defaultdict

def count_words(input_file):
    word_count = defaultdict(int)
    with open(input_file, 'r') as file:
        for line in file:
            words = re.findall(r'\w+', line.lower())  # Extract words and ignore non-word characters
            for word in words:
                word_count[word] += 1
    return word_count

def main():
    if len(sys.argv) < 2:
        print("Usage: python word_count_simple.py <input-file>")
        return

    input_file = sys.argv[1]
    word_count = count_words(input_file)

    # Print word counts
    for word, count in word_count.items():
        print(f"{word}: {count}")

if __name__ == "__main__":
    main()

File: CI Codes/MapReduce/Without Hadoop/WordCount.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class WordCount{

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java WordCountSimple <input-file>");
            return;
        }

        String inputFile = args[0];
        Map<String, Integer> wordCount = new HashMap<>();

        try (BufferedReader br = new BufferedReader(new FileReader(inputFile))) {
            String line;
            while ((line = br.readLine()) != null) {
                line = line.toLowerCase();
                String[] words = line.split("\\W+"); // Split by non-word characters
                for (String word : words) {
                    if (!word.isEmpty()) {
                        wordCount.put(word, wordCount.getOrDefault(word, 0) + 1);
                    }
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        // Print word counts
        for (Map.Entry<String, Integer> entry : wordCount.entrySet()) {
            System.out.println(entry.getKey() + " : " + entry.getValue());
        }
    }
}